mkosi/config.py

```diff
@@ -3785,21 +3785,33 @@ def parse_config_one(self, path: Path, profiles: bool = False, local: bool = Fal
         extras = path.is_dir()
 
         if path.is_dir():
-            path = path / "mkosi.conf"
+            path /= "mkosi.conf"
 
         if not self.match_config(path):
             return False
 
         if extras:
-            if local and (path.parent / "mkosi.local.conf").exists():
-                self.parse_config_one(path.parent / "mkosi.local.conf")
+            if local:
+                for p in (Path("mkosi.local"), Path("mkosi.local.conf")):
+                    localpath = path.parent / p
 
-                # Configuration from mkosi.local.conf should override other file based configuration but not
-                # the CLI itself so move the finalized values to the CLI namespace.
-                for s in SETTINGS:
-                    if hasattr(self.config, s.dest):
-                        setattr(self.cli, s.dest, self.finalize_value(s))
-                        delattr(self.config, s.dest)
+                    if localpath.is_dir():
+                        localpath /= "mkosi.conf"
```

Comment:
Looking at the code both `mkosi.local` and `mkosi.local.conf` can be either a file or directory, that seems a bit redundant. I don't think I understand the rationale.

Reply:
Huh right let me fix that